File: CodeGeneratorMVC/oldClasses/DALClass.vb
Option Strict On

Imports System.ComponentModel

<Serializable()> Public Class DALClass
    Implements INotifyPropertyChanged
    Private _ReadOnlyConnectionString As ConnectionString
    Private _EditOnlyConnectionSTring As ConnectionString
    Private _NamespaceName As ProjectVariable
    Private _ID As Integer
    Private _Name As String = ""
    Public Property ReadOnlyConnectionString() As ConnectionString
        Get
            Return _ReadOnlyConnectionString
        End Get
        Set(ByVal value As ConnectionString)
            _ReadOnlyConnectionString = value
            Me.NotifyPropertyChanged("ReadOnlyConnectionString")
        End Set
    End Property
    Public Property EditOnlyConnectionstring() As ConnectionString
        Get
            Return _EditOnlyConnectionSTring
        End Get
        Set(ByVal value As ConnectionString)
            _EditOnlyConnectionSTring = value
            Me.NotifyPropertyChanged("EditOnlyConnectionString")
        End Set
    End Property
    Public Property NameSpaceName() As ProjectVariable
        Get
            Return _NamespaceName

        End Get
        Set(ByVal value As ProjectVariable)
            _NamespaceName = value
        End Set
    End Property

    Public Property ID() As Integer
        Get
            Return _ID
        End Get
        Set(ByVal value As Integer)
            _ID = value
        End Set
    End Property
    Public Property Name() As String
        Get
            Return _Name
        End Get
        Set(ByVal value As String)
            _Name = value
        End Set
    End Property
    Public ReadOnly Property Self() As DALClass
        Get
            Return Me
        End Get
    End Property
    Public Event PropertyChanged(ByVal sender As Object, ByVal e As System.ComponentModel.PropertyChangedEventArgs) Implements System.ComponentModel.INotifyPropertyChanged.PropertyChanged
    Private Sub NotifyPropertyChanged(ByVal name As String)
        RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(name))
    End Sub
    Public Overrides Function ToString() As String
        Return Name
    End Function

End Class

File: CodeGeneratorMVC/Models/DALClass.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Security;
using System.Text;
using System.Threading.Tasks;
using Microsoft.VisualBasic;
using System.ComponentModel;

[Serializable()]
public class DALClass{ //: INotifyPropertyChanged {
    private ConnectionString _ReadOnlyConnectionString;
    private ConnectionString _EditOnlyConnectionSTring;
    private ProjectVariable _NamespaceName;
    private int _ID;
    private string _Name = "";
    public ConnectionString ReadOnlyConnectionString {
        get {
            return _ReadOnlyConnectionString;
        }
        set {
            _ReadOnlyConnectionString = value;
            this.NotifyPropertyChanged("ReadOnlyConnectionString");
        }
    }
    public ConnectionString EditOnlyConnectionstring {
        get {
            return _EditOnlyConnectionSTring;
        }
        set {
            _EditOnlyConnectionSTring = value;
            this.NotifyPropertyChanged("EditOnlyConnectionString");
        }
    }
    public ProjectVariable NameSpaceName {
        get {
            return _NamespaceName;
        }
        set {
            _NamespaceName = value;
        }
    }

    public int ID {
        get {
            return _ID;
        }
        set {
            _ID = value;
        }
    }
    public string Name {
        get {
            return _Name;
        }
        set {
            _Name = value;
        }
    }
    public DALClass Self {
        get {
            return this;
        }
    }
    public event PropertyChangedEventHandler PropertyChanged;

    public delegate void PropertyChangedEventHandler(object sender, System.ComponentModel.PropertyChangedEventArgs e);

    private void NotifyPropertyChanged(string name) {
        PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(name));
    }
    public override string ToString() {
        return Name;
    }
}
